src/tir/transforms/lower_custom_datatypes.cc

```diff
@@ -97,7 +97,7 @@ class CustomDatatypesLowerer : public StmtExprMutator {
     return expr;
   }
 
-#define DEFINE_MUTATE__(OP, NodeName)                                              \
+#define DEFINE_MUTATE(OP, NodeName)                                                \
```

Comment:
What is the warning does this macro leads to? Seems the original code is fine as it is(with underscores as part of the macro)

Reply:
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2011/n3242.pdf section 17.6.4.3.2
For global names:
Each name that contains a double underscore _ _ or begins with an underscore followed by an uppercase letter (2.12) is reserved to the implementation for any use.
I never can remember these rules so to simplify things I never start any name with underscore and never use double underscores. There are cases where it is fine but I find it too difficult for me to remember when.
This define is within a namespace so I think it might be ok, but why tempt fate?